### PR TITLE
Handle Nexus links

### PR DESCRIPTION
**Pull request by rodrigozhou**
## What changed?

Handle Nexus links

## Why?

Support Nexus bi-directional links.

### CODE REVIEW

File: components/nexusoperations/executors.go
Comment (bergundy) [inline components/nexusoperations/executors.go:340]: I'd probably just log an error and skip this link in the response just to be safe. Alternatively you can ensure that this translates into a non retryable error but I would rather be more forgiving than break the operation.

File: service/frontend/nexus_handler.go
Comment (bergundy) [inline service/frontend/nexus_handler.go:327]: This is fine, I was more concerned about the response.

**Review by bergundy — COMMENTED** [review]
This mostly LGTM. Approving with a couple of minor comments.

**Review by bergundy — APPROVED** [review]

**Comment by bergundy** [comment]
And please add functional tests to confirm the links are properly sent everywhere.

File: service/frontend/nexus_handler.go
Comment (bergundy) [inline service/frontend/nexus_handler.go:321]: Let's not decode here, the SDK will have to decode anyways.

File: components/nexusoperations/executors.go
Comment (bergundy) [inline components/nexusoperations/executors.go:332]: We should either ignore or fail the operation. Returning an error here would cause us to retry.

Comment (bergundy) [reply components/nexusoperations/executors.go:332]: Let's ignore for simplicity now.

File: components/nexusoperations/link_converter.go
Comment (bergundy) [inline components/nexusoperations/link_converter.go:37]: Technically, you don't need to parse here, you could construct the URL directly but this is also fine.

File: components/nexusoperations/link_converter.go
Comment (bergundy) [inline components/nexusoperations/link_converter.go:41]: Don't these need to be escaped?

File: components/nexusoperations/link_converter.go
Comment (bergundy) [inline components/nexusoperations/link_converter.go:58]: Why the extra `.`?

File: components/nexusoperations/link_converter.go
Comment (bergundy) [inline components/nexusoperations/link_converter.go:93]: AFAIR we use camel case in our HTTP API for query params, let's make sure we're consistent. I will double check.

File: components/nexusoperations/link_converter.go
Comment (bergundy) [inline components/nexusoperations/link_converter.go:78]: Why parse the query? You have it parsed already.

File: components/nexusoperations/link_converter.go
Comment (bergundy) [inline components/nexusoperations/link_converter.go:1]: We need some unit tests for this.

File: components/nexusoperations/link_converter.go
Comment (rodrigozhou) [reply components/nexusoperations/link_converter.go:41]: When `url.URL.String()` is called, they are escaped.

File: components/nexusoperations/link_converter.go
Comment (rodrigozhou) [reply components/nexusoperations/link_converter.go:58]: Just in case someone defines another type with this one as prefix.

File: components/nexusoperations/link_converter.go
Comment (rodrigozhou) [reply components/nexusoperations/link_converter.go:1]: I wrote them in the SDK. Gonna remove this file later.

File: components/nexusoperations/link_converter.go
Comment (rodrigozhou) [reply components/nexusoperations/link_converter.go:78]: What do you mean? This is parsing the event reference.

File: components/nexusoperations/link_converter.go
Comment (bergundy) [reply components/nexusoperations/link_converter.go:41]: What if they contain slashes?

File: components/nexusoperations/link_converter.go
Comment (bergundy) [reply components/nexusoperations/link_converter.go:58]: Not following... we only support the exact fullname right now, you can check equality instead of prefix match.

File: components/nexusoperations/link_converter.go
Comment (bergundy) [reply components/nexusoperations/link_converter.go:78]: I mean that you pass `RawQuery` instead of `Query`

File: components/nexusoperations/link_converter.go
Comment (bergundy) [reply components/nexusoperations/link_converter.go:1]: Can you put them here for completeness please?

File: components/nexusoperations/link_converter.go
Comment (rodrigozhou) [reply components/nexusoperations/link_converter.go:58]: So, because we also event reference, we need to add the event reference type to there. So, it's actual something like this: `temporal.api.common.v1.Link.WorkflowEvent.EventReference`

File: components/nexusoperations/link_converter.go
Comment (rodrigozhou) [reply components/nexusoperations/link_converter.go:78]: Oh, well. `url.Parse` doesn't really check if the query string is valid. It silently ignore keys/values that cannot be parsed.

File: components/nexusoperations/link_converter.go
Comment (rodrigozhou) [reply components/nexusoperations/link_converter.go:41]: Oh, good point! Let me fix it!

File: components/nexusoperations/link_converter.go
Comment (bergundy) [reply components/nexusoperations/link_converter.go:78]: That's good, we can ignore those too.

File: components/nexusoperations/link_converter.go
Comment (bergundy) [reply components/nexusoperations/link_converter.go:93]: Looks like it from one of the tests:

```
		"/namespaces/"+s.namespace+"/workflows/"+workflowID+"/history?historyEventFilterType=2",
```

File: components/nexusoperations/link_converter.go
Comment (bergundy) [reply components/nexusoperations/link_converter.go:58]: The type should be `temporal.api.common.v1.Link.WorkflowEvent` and we should be able to infer the reference type from the query params. Maybe add a query param that says it's an event reference?

File: components/nexusoperations/executors.go
Comment (bergundy) [inline components/nexusoperations/executors.go:339]: As discussed the type should be WorkflowEvent.

File: components/nexusoperations/link_converter_test.go
Comment (bergundy) [inline components/nexusoperations/link_converter_test.go:39]: We need a test for workflow ID with a `/`.

File: components/nexusoperations/link_converter.go
Comment (bergundy) [inline components/nexusoperations/link_converter.go:65]: We should also check the that the protocol is `temporal`.

File: components/nexusoperations/executors.go
Comment (bergundy) [inline components/nexusoperations/executors.go:342]: Would have been better to log so there's indication but let's do that in a follow up PR.

File: components/nexusoperations/executors_test.go
Comment (bergundy) [inline components/nexusoperations/executors_test.go:371]: nit: you could have just merged this into `onStartOperation`. Not blocking the PR though.

**Review by bergundy — APPROVED** [review]
Good to merge.

**Pull request merged**
